bitflow/sample.go

```diff
@@ -45,6 +45,9 @@ func (v Value) String() string {
 type Header struct {
 	// Fields defines the names of the metrics of samples belonging to this header.
 	Fields []string
+
+	// Buffer for checking of contained field values
+	buffer map[string]bool
```

Comment:
See my comment below about the ContainsField method. The BuildIndex() method provides a map similar to the new `buffer` field, where the presence of a header field can be checked using this map syntax: `index, exists := header.BuildIndex()["a"]`. Unless I am missing something, I don't think the changes in sample.go are necessary.

Reply:
Done